src/main/java/com/flightstats/hub/channel/ChannelContentResource.java

```diff
@@ -225,6 +226,81 @@ private Response getTimeQueryResponse(String channel, DateTime startTime, String
         }
     }
 
+    @Path("/{h}/{m}/{second}/{direction:[n|p].*}/{count}")
+    @Produces({MediaType.APPLICATION_JSON})
+    @GET
+    public Response getDirectionalSecond(@PathParam("channel") String channel,
+                                         @PathParam("Y") int year,
+                                         @PathParam("M") int month,
+                                         @PathParam("D") int day,
+                                         @PathParam("h") int hour,
+                                         @PathParam("m") int minute,
+                                         @PathParam("second") int second,
+                                         @PathParam("count") int count,
+                                         @PathParam("direction") String direction,
+                                         @QueryParam("location") @DefaultValue(Location.DEFAULT) String location,
+                                         @QueryParam("epoch") @DefaultValue(Epoch.DEFAULT) String epoch,
+                                         @QueryParam("trace") @DefaultValue("false") boolean trace,
+                                         @QueryParam("stable") @DefaultValue("true") boolean stable,
+                                         @QueryParam("order") @DefaultValue(Order.DEFAULT) String order) {
+        SortedSet<ContentPath> keys = new TreeSet<>();
+        DateTime stableTime = TimeUtil.stable();
+        DateTime startTime = new DateTime(year, month, day, hour, minute, second, 999, DateTimeZone.UTC);
+        boolean next = direction.startsWith("n");
+        if (next) {
+            while (startTime.isBefore(stableTime) && keys.size() < count) {
```

Comment:
Am I reading this correctly: this just returns URLs to the time API, not links to the data items?

Reply:
Yup. This is an example payload:

```
{
  "_links" : {
    "self" : {
      "href" : "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/45/n/2"
    },
    "next" : {
      "href" : "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/46/next/2"
    },
    "previous" : {
      "href" : "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/45/previous/2"
    },
    "uris" : [
      "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/45",
      "http://hub.dls.dev.flightstats.io/channel/single_test_1/2018/08/08/18/06/46"
    ]
  }
}
```